packages/ovm/src/contracts/FullStateManager.sol

```diff
@@ -123,10 +124,28 @@ contract FullStateManager is StateManager {
      * @param _codeContractAddress The address of the code contract.
      * @return The hash of the bytecode at this address.
      */
-    function getCodeContractHash(address _codeContractAddress) internal view returns (bytes32 _codeContractHash) {
+    function getCodeContractHash(address _codeContractAddress) public view returns (bytes32 _codeContractHash) {
         // TODO: Look up cached hash values eventually to avoid having to load all of this bytecode
         bytes memory codeContractBytecode = getCodeContractBytecode(_codeContractAddress);
         _codeContractHash = keccak256(codeContractBytecode);
         return _codeContractHash;
     }
+
+    /**
+     * @notice Deploys a code contract, and then registers it to the state
+     * @param _ovmContractInitcode The bytecode of the contract to be deployed
+     * @return the codeContractAddress.
+     */
+    function deployCodeContract(bytes memory _ovmContractInitcode) public returns(address codeContractAddress) {
```

Comment:
Oooo so this function unfortunately needs to be a little more complex. Instead of `deployCodeContract(_ovmContractInitcode)` what we really need is `deployContract(_ovmContractAddress, _ovmContractInitcode)`

This function will also need to inherit the logic from here: https://github.com/ethereum-optimism/optimism-monorepo/blob/c5263562d165495e3437d25e83f3a808d1f395e3/packages/ovm/src/contracts/ExecutionManager.sol#L586-L595 where we associate a code contract with it.

The reason is we want to be able to intercept this call and actually deploy a contract (with the initcode) to the Geth state at the address which the ExecutionManager wants us to deploy it to.

Reply:
Ah, yep! [Done](https://github.com/ethereum-optimism/optimism-monorepo/pull/132/commits/5a161c6e6ec9f15f9deb80879fc2d6ce656b2060)!